test/unit/Container.spec.ts

```diff
@@ -128,13 +112,13 @@ describe("Container", function() {
             }
         }
 
-        function createCar (engine: Engine) {
-            return new Car(engine);
+        class CarFactory {
+            @Factory()
+            public static createCar (engine: Engine): Car {
```

Comment:
Okay, I'll remove `@Factory`. Considering the other question — the problem is, we can't decide for the user. She could have a preexisting system with factories defined in one of these three ways (or in mixed ways). The container should be able to handle all cases the same. What do you think about the third case? How should we pass the factory class as well as a factory function to the container?

Actually, there is another case — when factory is already registered with the container using named service. We should also allow a way to reference the factory using a service name. The API could look like this:

```ts
@Service({ factory: ["car-factory", CarFactory.prototype.createCar] }) class Car {}
                                         ^--- Name of the service registered with the container
```